collections/motoko/dfx.json

```diff
@@ -1,24 +1,40 @@
 {
   "canisters": {
     "hashmap": {
-      "type": "motoko",
-      "main": "src/hashmap.mo"
+      "type": "custom",
+      "candid": "../rust/collection.did",
+      "wasm": "hashmap.wasm",
+      "build": [
+        "sh -c '$(dfx cache show)/moc src/hashmap.mo -o hashmap.wasm --public-metadata candid:service --package base $(dfx cache show)/base'"
+      ]
     },
-    "triemap" : {
-      "type": "motoko",
-      "main": "src/triemap.mo"
+    "triemap": {
+      "type": "custom",
+      "candid": "../rust/collection.did",
+      "wasm": "triemap.wasm",
+      "build": [
+        "sh -c '$(dfx cache show)/moc src/triemap.mo -o triemap.wasm --public-metadata candid:service --package base $(dfx cache show)/base'"
+      ]
     },
-    "rbtree" : {
-      "type": "motoko",
-      "main": "src/rbtree.mo"
+    "rbtree": {
+      "type": "custom",
+      "candid": "../rust/collection.did",
+      "wasm": "rbtree.wasm",
+      "build": [
+        "sh -c '$(dfx cache show)/moc src/rbtree.mo -o rbtree.wasm --public-metadata candid:service --package base $(dfx cache show)/base'"
```

Comment:
I think we can use the default `motoko` type without a custom build?

Reply:
I tried that, but it actually uses the `base` from vessel, because `vessel` is still used by Zhenya's hashmap and the Btree, which transitively depends on base.

Reply:
If we do something like this: https://github.com/dfinity/motoko-playground/blob/main/package-set.dhall#L10, would it override the transitive dependency? It's best if we can use the default motoko type, so that changes from `dfx build` can be propagated here.

Reply:
I think so, although it's possible that the BTree for example relies on a specific version of base that is different from what we are using.

I'm trying out a `mops` approach that looks like it might work.